app/[lang]/types.d.ts

```diff
@@ -0,0 +1,15 @@
+interface PageProps {
```

Comment:
This is quite subjective, but I prefer creating the types in the file they are used instead of creating `types.d.ts`-files